scratch/evaluate_gaming_traffic.cc

```diff
@@ -0,0 +1,148 @@
+      interArrivalTimes.push_back (difference.GetSeconds ());
+      lastPacketTime = packetTime;
+    }
+
+  packetsSize.push_back (packet->GetSize ());
+}
+
+void
+CalculateThroughput ()
+{
+  Time now = Simulator::Now ();    /* Return the simulator's virtual time. */
+  double cur = (packetSink->GetTotalRx () - lastTotalRx) * 8.0 / 1e3;     /* Convert Application RX Packets to kb. */
+  std::cout << now.GetSeconds () << "s: \t" << cur << " kbps" << std::endl;
+  lastTotalRx = packetSink->GetTotalRx ();
+  Simulator::Schedule (MilliSeconds (100), &CalculateThroughput);
+}
+
+int
+main (int argc, char *argv[])
+{
+  bool verbose = false;  /* print logging information */
+  bool summary = true;  /* print application layer traffic summary */
+  bool throughput = false;  /* print system throughput */
+  double simulationTime = 10.0;  /* simulation time in seconds */
+  std::string gameId = "ns3::CrazyTaxiStreamingServer"; /* TypeId of the game */
+
+  CommandLine cmd;
+  cmd.AddValue ("verbose", "Print the logging information", verbose);
+  cmd.AddValue ("summary", "Print summary of application layer traffic", summary);
+  cmd.AddValue ("throughput", "Print the system throughput", throughput);
+  cmd.AddValue ("time", "Simulation time (in Seconds)", simulationTime );
+  cmd.AddValue ("gameId", "The game TypeId", gameId );
+  cmd.Parse (argc, argv);
+
+  Time::SetResolution (Time::NS);
+  if (verbose)
+    {
+      LogComponentEnable ("GamingStreamingServer", LOG_LEVEL_INFO);
+      LogComponentEnable ("PacketSink", LOG_LEVEL_INFO);
+    }
+
+  NodeContainer nodes;
+  nodes.Create (2);
+
+  PointToPointHelper pointToPoint;
+  pointToPoint.SetDeviceAttribute ("DataRate", StringValue ("50Mbps"));
+  pointToPoint.SetChannelAttribute ("Delay", StringValue ("2ms"));
+
+  NetDeviceContainer devices;
+  devices = pointToPoint.Install (nodes);
+
+  InternetStackHelper stack;
+  stack.Install (nodes);
+
+  Ipv4AddressHelper address;
+  address.SetBase ("10.1.1.0", "255.255.255.0");
+
+  Ipv4InterfaceContainer interfaces = address.Assign (devices);
+
+  GamingStreamingServerHelper gamingStreamingHelper (gameId, interfaces.GetAddress (0), 9);
+
+  ApplicationContainer serverApps = gamingStreamingHelper.Install (nodes.Get (1));
+  gamingServer = StaticCast<GamingStreamingServer> (serverApps.Get (0));
+  serverApps.Start (Seconds (1.0));
+  serverApps.Stop (Seconds (simulationTime + 1.0));
+  /** End of Streaming Server*/
+
+  PacketSinkHelper sink ("ns3::UdpSocketFactory", InetSocketAddress (Ipv4Address::GetAny (), 9));
+  ApplicationContainer sinkApps = sink.Install (nodes.Get (0));
+  packetSink = StaticCast<PacketSink> (sinkApps.Get (0));
+  sinkApps.Start (Seconds (0.0));
+  sinkApps.Stop (Seconds (simulationTime + 1.1));
+  if (throughput)
+    {
+      Simulator::Schedule (Seconds (1.1), &CalculateThroughput);
+    }
+  packetSink->TraceConnectWithoutContext ("Rx", MakeCallback (&ReceivePacket));
+
+  Simulator::Stop (Seconds (12.1));
```

Comment:
The simulation stop time should be set according to `simulationTime`.
Please take a look at other evaluation scenarios